wgpu-hal/src/vulkan/mod.rs

```diff
@@ -1210,6 +1211,12 @@ impl crate::Queue for Queue {
             signal_values.push(!0);
         }
 
+        let mut pending_signal_semaphores = self.signal_semaphores.lock();
+        if !pending_signal_semaphores.is_empty() {
+            signal_values.append(vec![!0; pending_signal_semaphores.len()].as_mut());
```

Comment:
```suggestion
            signal_values.extend(iter::repeat(!0).take(pending_signal_semaphores.len()));
```
Something like that avoids allocation entirely.